media_driver/linux/common/os/mos_utilities_specific.c:2617:12: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
    return (uint32_t)pthread_self();
           ^~~~~~~~~~~~~~~~~~~~~~~~

--- media_driver/linux/common/os/mos_utilities_specific.c.orig	2018-10-01 21:39:52 UTC
+++ media_driver/linux/common/os/mos_utilities_specific.c
@@ -2614,7 +2614,7 @@ uint32_t MOS_GetThreadId(
 
 uint32_t MOS_GetCurrentThreadId()
 {
-    return (uint32_t)pthread_self();
+    return (uintptr_t)pthread_self();
 }
 
 MOS_STATUS MOS_WaitThread(
